money.go

```diff
@@ -27,12 +27,34 @@ func defaultUnmarshalJSON(m *Money, b []byte) error {
 	if err != nil {
 		return err
 	}
-	ref := New(int64(data["amount"].(float64)), data["currency"].(string))
+
+	amountRaw := data["amount"]
+	var amount float64
+	if amountRaw != nil {
+		amount = amountRaw.(float64)
+	}
+	currencyRaw := data["currency"]
+	var currency string
+	if currencyRaw != nil {
+		currency = data["currency"].(string)
```

Comment:
do you think it is worthwhile to check and return an error if a given currency value is a `string` instead of panicking if a e.g. number is given?

Reply:
Hmm you mean return an error if it's _not_ a string? I think you're suggesting "return a JSON unmarshalling error instead of panicking if the currency or amount isn't the right type", which I think is a good idea. I can make that update is that what you mean?

Reply:
yes, you're right. To type check and return an error instead of panicking

Reply:
Ok cool, I've pushed a commit to address this!